extensions/eclipse-che-theia-git-provisioner/src/node/git-configuration-controller.ts

```diff
@@ -0,0 +1,127 @@
+/********************************************************************************
+ * Copyright (C) 2018-2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/// <reference types='@theia/core/src/typings/nsfw/index'/>
+
+import { injectable, inject } from 'inversify';
+
+import { homedir } from 'os';
+import { resolve } from 'path';
+import { CheTheiaUserPreferencesSynchronizer } from '@eclipse-che/theia-user-preferences-synchronizer/lib/node/che-theia-preferences-synchronizer';
+import { writeFile, pathExists, createFile, readFile } from 'fs-extra';
+import * as ini from 'ini';
+import * as nsfw from 'nsfw';
+import { Disposable } from '@theia/core';
+
+export const GIT_CONFIG_PATH = resolve(homedir(), '.gitconfig');
```

Comment:
It is better to have `path.resolve` here,  because resolve is often used in promises.